server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1089,10 +1089,6 @@ private ReplicationGroup calculateReplicationGroup() {
             newVersion = replicationGroup.getVersion() + 1;
         }
 
-        assert indexSettings().isRemoteTranslogStoreEnabled()
```

Comment:
Lets not remove this assertion. Removing this can pose risks to bugs.

We can check first if index is remote enabled, or if there is an inflight migration, or all tracked shards are replicated.